### DISCUSSION TITLE
How to response with empty body?

### DISCUSSION THREAD

**Discussion by symonsoft** [fastapi/fastapi#7819] [Questions] [answered]
Though I neither declare response model (None by default) neither return anything the response contains `null`. Is there a way to send empty body?

**Comment by phy25** [comment]
> ### First check
> 
> * [ ] I used the GitHub search to find a similar issue and didn't find it.
> * [ ] I searched the FastAPI documentation, with the integrated search.
> * [ ] I already searched in Google "How to X in FastAPI" and didn't find any information.

https://github.com/tiangolo/fastapi/issues/449#issuecomment-523371704

**Comment by alvarogf97** [comment]
Do you want to return HTTP_204_NO_CONTENT? If it is true, remove `return` statement from your endpoint handler.

**Comment by iishyfishyy** [comment]
I know in Flask you had to do something like `return '', 200`. You could try returning an empty string. There's probably a cleaner way, though.

**Comment by f0ff886f** [comment]
This may help: https://github.com/tiangolo/fastapi/issues/717#issuecomment-583826657

**Comment by symonsoft** [comment]
@alvarogf97 
> Do you want to return HTTP_204_NO_CONTENT? If it is true, remove `return` statement from your endpoint handler.

I want to return 200 and no body at all.
If I remove `return` response status is 200 and body is "null".

**Comment by symonsoft** [comment, marked as answer]
This works as I need:
`return Response(status_code=HTTP_200_OK)`

Thanks all!

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :cake:

Thanks @symonsoft for reporting back and closing the issue :+1:

**Comment by gabigabogabu** [comment]
imports for the answer from @symonsoft are

`from starlette.responses import Response`
`from starlette.status import HTTP_200_OK`

**Comment by remidinishanth-ntnx** [comment]
You can also import from fastapi instead of starlette using
```
from fastapi import Response, status
```
and use `status.HTTP_200_OK`

**Comment by teror4uks** [comment]
see https://github.com/tiangolo/fastapi/issues/717#issuecomment-753360668 as for me is the best solution

**Comment by halon176** [comment]
Is there a way to document a response without a body but with a 200 status code instead of 204? Otherwise, it automatically gets documented with a response body of type "string"

**Reply by Zlika** [reply]
I have the same problem: I can send an answer with no body, but the swagger-ui shows an example body content with "string". I cannot find any way to show in the swagger-ui that the body is empty.

**Reply by MajorDallas** [reply]
> see [#717 (comment)](https://github.com/fastapi/fastapi/issues/717#issuecomment-753360668) as for me is the best solution

That solution also addresses this issue.

The short version: set `response_class=Response` in the decorator.
